app/controllers/bulk_submission_excel/downloads_controller.rb

```diff
@@ -33,4 +35,17 @@ def create # rubocop:todo Metrics/AbcSize, Metrics/MethodLength
   def submission_parameters
     params.require(:bulk_submission_excel_download).permit(:asset_barcodes)
   end
+
+  private
+
+  # Build a filename for the file to be downloaded
+  # Consists of the template name in kebab-case, followed by the current date and time in ISO8601 format
+  # e.g. "submission-template_20190101T120000.xlsx"
+  def build_filename
+    template_name = params.dig(:defaults, :template_name) || 'Submission Template'
+    # convert to kebab-case, catching existing hyphens
+    template_name = template_name.downcase.gsub(/[-\s]+/, '-')
+    "#{template_name.dasherize}_#{Time.current.strftime('%Y%m%dT%H%M%S')}.xlsx"
```

Comment:
Although the comment says it is the current data and time in ISO8601 format, this part has no TZ info. Therefore, could you please add to the comment whether it is local or UTC so the intention is clear?

Reply:
I mean this only as a code comment as general when writing a timestamp, but it is OK in the filename too I think.